src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -365,9 +365,26 @@ private void ConfigureDeploymentFromConfigFile(Recommendation recommendation, Us
             foreach (var entry in userDeploymentSettings.LeafOptionSettingItems)
             {
                 var optionSettingJsonPath = entry.Key;
+                var optionSettingKey = string.Empty;
                 var optionSettingValue = entry.Value;
 
-                var optionSetting = recommendation.GetOptionSetting(optionSettingJsonPath);
+                OptionSettingItem? optionSetting = null;
+                try
+                {
+                    optionSetting = recommendation.GetOptionSetting(optionSettingJsonPath);
+                }
+                catch (OptionSettingItemDoesNotExistException)
+                {
+                    var optionSettingSplit = optionSettingJsonPath.Split(".");

```

Comment:
This needs a comment. I have no idea what this code is here for.

Reply:
From what I can gather the logic attempt to get an `optionSetting`. If it fails then it is likely a KeyValuePair and do some extra logic for for determing `optionSettingKey`.

Its not good to use control flow by exceptions and having a optionSettingKey variable that is there for all types but only used if it is a KVP adds even more confusion to the code. This method is getting to frail relying on subtle behavior with nothing documented.

Reply:
This code has been revised and moved to GetOptionSetting